### PR TITLE
feat(frontend): chore create quantity component

### PR DESCRIPTION
**Pull request by markjoshuacarmen**
## Ticket/s

- [Increase/Decrease Item Quantity Component](https://trello.com/b/FeFcZk4b/eycommerce/power-ups)

## Changes

- chore: create quantity component

## Screenshot

- image

## How to test?

- checkout to branch
- changes temporarily placed in SampleView
- access in browser http://localhost:5173/
note: will remove the component in SampleView before merge once approved

### CODE REVIEW

File: packages/ui/quantity/Quantity.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import { Count, QuantityWrapper, SquareButton } from './style';
+
+export function Quantity() {
+  const [count, handlers] = useCounter(0, { min: 0, max: 100 });
+
+  return (
+    <QuantityWrapper>
+      <SquareButton onClick={handlers.decrement}>
+        <BsDash style={{ margin: '0', padding: '0' }} />
```
Comment (SGVGen) [inline packages/ui/quantity/Quantity.tsx:12]: import { BsDashStyled, Count, QuantityWrapper, SquareButton } from './style';

File: packages/ui/quantity/Quantity.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      </SquareButton>
```
Comment (SGVGen) [inline packages/ui/quantity/Quantity.tsx]: Inline Styling - Try this instead:
```suggestion
      import { BsDashStyled } from './style';
      <SquareButton onClick={handlers.decrement}>
        <BsDashStyled />
      </SquareButton>
```

File: packages/ui/quantity/style.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  border: 1px solid #ccc;
+  margin: 0;
+  padding: 0;
+`;
+
+export const Count = styled.div`
+  ${SquareButton}
+  margin: 0;
+  padding: 0;
+`;
```
Comment (SGVGen) [inline packages/ui/quantity/style.ts:38]:
```suggestion
import { BsDash } from 'react-icons/bs';
export const BsDashStyled = styled(BsDash)`
  margin: 0;
  padding: 0;
`;
```

File: packages/ui/quantity/Quantity.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { useCounter } from '@mantine/hooks';
+import { BsDash, BsPlus } from 'react-icons/bs';
+
+import { Count, QuantityWrapper, SquareButton } from './style';
+
+export function Quantity() {
+  const [count, handlers] = useCounter(0, { min: 0, max: 100 });
```
Comment (brentmillen) [inline packages/ui/quantity/Quantity.tsx]: as discussed, it won't be allowed to place 0 quantity. try changing the min value to "1" for this mantine hook.

**Review by brentmillen — CHANGES_REQUESTED** [review]

File: packages/ui/quantity/Quantity.tsx (hunk 4 above)
Comment (markjoshuacarmen) [reply packages/ui/quantity/Quantity.tsx]: ok sir

**Review by brentmillen — APPROVED** [review]

File: packages/ui/quantity/Quantity.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+export function Quantity() {
+  const [count, handlers] = useCounter(0, { min: 1, max: 100 });
+
+  return (
+    <QuantityWrapper>
+      <SquareButton onClick={handlers.decrement}>
+        <BsDashStyled />
+      </SquareButton>
+      <SquareButton>
+        <Count>{count}</Count>
```
Comment (brentmillen) [inline packages/ui/quantity/Quantity.tsx]: Allow input of number for quantity

Comment (brentmillen) [reply packages/ui/quantity/Quantity.tsx]: also handle number only input for this change

Comment (markjoshuacarmen) [reply packages/ui/quantity/Quantity.tsx]: done

**Review by sandrex-SGV — APPROVED** [review]
Lufet
LGTM

**Pull request merged**